packages/components/Callout/macos/Callout.swift

```diff
@@ -281,37 +270,53 @@ class CalloutView: RCTView, CalloutWindowLifeCycleDelegate {
 
 	// MARK: Private variables
 
+	/// The view the Callout is presented from, if anchorRect is nil.
 	private var anchorView: NSView?
 
-	private var proxyView = RCTView()
+	/// The  view we forward Callout's Children to. It's hosted within the CalloutWindow's
+	/// view heirarchy, ensuring our React Views are not placed in the main window.
+	private lazy var proxyView: NSView = {
+		let visualEffectView = FlippedVisualEffectView()
+		visualEffectView.translatesAutoresizingMaskIntoConstraints = false
+		visualEffectView.material = .menu
+		visualEffectView.state = .active
+		visualEffectView.wantsLayer = true
+		visualEffectView.layer?.cornerRadius = calloutWindowCornerRadius
+
+		/**
+		 * We can't directly call touchHandler.attach(to:) because `visualEffectView` is not an RCTUIView.
+		 * We get around this limitation by just replicating what `attach` did internally: add a gestureRecognizer.
+		 */
+		guard let touchHandler = RCTTouchHandler(bridge: bridge) else {
+			preconditionFailure("Callout could not create RCTTouchHandler")
+		}
+		visualEffectView.addGestureRecognizer(touchHandler)
+
+		return visualEffectView
+	}()
 
 	private lazy var calloutWindow: CalloutWindow = {
 		let window = CalloutWindow()
 		window.lifeCycleDelegate = self
 
-		let visualEffect = NSVisualEffectView()
-		visualEffect.translatesAutoresizingMaskIntoConstraints = false
-		visualEffect.material = .menu
-		visualEffect.state = .active
-		visualEffect.wantsLayer = true
-		visualEffect.layer?.cornerRadius = calloutWindowCornerRadius
-
 		guard let contentView = window.contentView else {
 			preconditionFailure("Callout window has no content view")
 		}
 
-		contentView.addSubview(visualEffect)
```

Comment:
dumb question : why is it ok to remove the constraints on proxyview?

Reply:
Before the view hierarchy looked like NSWindow->ContentView->VisualEffectView->proxyView.
Turns out the visualEffectView could just be the proxyView, so I didn't need the extra `RCTView`. And also, we no longer need those constraints.. somehow.

Reply:
you are setting visualEffectView.translatesAutoresizingMaskIntoConstraints = false .... so who is setting the constraint? have you look at ui debugger to see if there are any ambiguous size?

Reply:
There seems to not be. It looks like it inherited the size just fine. In general, there is no set size in the whole view hierarchy, so I think it's React Native is doing it's own layout pass where it is setting the frame on the Callout, which get's forwarded to the the proxyView frame (visualEffectView)

![image](https://user-images.githubusercontent.com/6722175/151264804-b6377049-4bb4-4582-857f-b0a09098ce40.png)

Reply:
I thought about it more and have an answer. Now that `FlippedVisualEffectView` _is_ the proxy view, it's frame is set by React Native on line 69. React Native uses Yoga / Shadow Views to do Layout, so that's why we don't need constraints.